server.go

```diff
@@ -1,66 +0,0 @@
-package main
-
-import (
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"net/http"
-	"log"
-)
-
-// *** Structs *** 
-
-type User struct {
-	UserID int 
-	Emails []string 
-}
-
-type Payload struct {
-	UserData User 
-}
-
-// **************
-
-// func serveRest(w http.ResponseWriter, r *http.Request) {
-// 	response, err := getJsonResponse()
-// 	if err != nil {
-// 		panic(err)
-// 	}
-
-// 	fmt.Fprintf(w, string(response))
-// }
-
-func retrieve(w http.ResponseWriter, r *http.Request) {
-	//url := "http://localhost:8081"
-	// res, err := http.Get(url)
-	// if err != nil {
-	// 	panic(err)
-	// }
-	// defer res.Body.Close()
-
-	body, err := ioutil.ReadAll(r.Body)
-	
-	if err != nil {
-		panic(err)
-	}
-
```

Comment:
I see `defer r.Body.Close()`  is the last line. It's a good thing to always close the body, however "A defer statement defers the execution of a function until the surrounding function returns." So as the last statement the `defer` keyword is really not needed. If you moved it up to line 46 it would be better utilized, and its generally a good idea from the standpoint of organizing code, because the `Close()` statement is next to the `ReadAll()` statement. See: https://gobyexample.com/defer